Android Version (beta)

patw0929 opened the pull request:
As mentioned at https://github.com/gitpoint/git-point/issues/2#issuecomment-315032689, I've made a Android version. What I do were replacing the components `ActionSheetIOS` & `react-native-search-bar`, because which are only compatible with iOS.

I think there might be some bugs, and I need other contributors help to check the features are works like expected (and before) on Android & iOS.

Thanks for reviewing this :)
Very looking forward to see the Android version in the Google Play.

housseindjirdeh commented:
This is an **outstanding** PR 🎉 🎉 

I can't thank you enough for this. I'm going to take a good look at this tomorrow morning - test it on an Android device and we'll get it merged in. Once it's in, I'll spend some time going through the app  to make sure things are smooth (there may definitely be some more things here and there) as well as check for regression. I'm so excited that Android looks like it'll be ready *a lot* sooner than I imagined and it's all thanks to you and everyone else assisting <3

alejandronanez commented on App.js:
```diff
@@ -40,8 +46,10 @@ class App extends Component {
       secretKey: md5(DeviceInfo.getUniqueID())
     });
 
+    store = configureStore();
```
I'm curious, any specific reason to do this in here instead inside `persistStore(...)`?

alejandronanez commented on package.json:
```diff
@@ -21,10 +22,12 @@
     "react-native-device-info": "^0.10.2",
     "react-native-elements": "^0.9.3",
     "react-native-htmlview": "^0.9.0",
+    "react-native-material-design-searchbar": "^1.1.4",
```
Any reason to add this dep in here? We already have a search bar in `react-native-elements` https://react-native-training.github.io/react-native-elements/API/searchbar/

housseindjirdeh replied:
Yep we've discussed this in #2. So you'll notice that I already use `react-native-search-bar` instead of the RNE solution. That library provides a wrapper around a native iOS search bar instead of relying on a non-native solution.

We've discussed that it probably makes sense to use a separate library for Android and keep the iOS logic as native as possible. However I have two questions:

1. Is there any sort of actual native Android search bar with a react-native bridge we can use instead?
2. If 1 doesn't seem to exist - then that's okay. The last time I tried using the search bar with RNE - it wasn't the simplest to modify and use. I don't know if that's changed however. I think it's worth a quick check and if we think that using this imported library than I'm okay with that

alejandronanez replied:
Is there something like an universal solution for search bar? I don't think it's a good idea to have split specific components for each platform that makes our code less mantainable. Also we don't want our search bar to depend on 2 different projects.
Hopefully that makes sense.

alejandronanez replied:
Hey @patw0929! thanks for the quick changes!
So according to your (comment)[https://github.com/gitpoint/git-point/pull/49#issuecomment-315505647] I'm glad we should use une search bar solution that we can use in both iOS and Android. Do you have any suggestions?

patw0929 replied:
I think I will try react-native-elements' search bar later. Another choice might be the react-native-search-bar.

alejandronanez commented on src/components/search-bar.component.ios.js:
```diff
@@ -0,0 +1,46 @@
+import React, { Component, PropTypes } from 'react';
+import RNSearchBar from 'react-native-search-bar';
```
I have the same question, since we're using `react-native-elements` I think we should stick with that instead of start adding more deps into the project. You can find the component of `react-native-elements` in here react-native-search-bar

housseindjirdeh replied:
See my comment above too!

alejandronanez commented on src/issue/screens/pull-merge.screen.js:
```diff
@@ -68,9 +68,22 @@ class PullMerge extends Component {
+  };
+
+  handlePress = (i) => {
+    if (i !== this.mergeMethodMessages.length) {
```
What does `i` stand out for? I think it doesn't hurt to add a more descriptive name in here.

alejandronanez replied:
Something like `index` or `issue` for example.

alejandronanez commented on src/user/screens/profile.screen.js:
```diff
@@ -84,10 +80,11 @@ class Profile extends Component {
       isPendingOrgs,
       isPendingCheckFollowing,
-      navigation
+      navigation,
```
Remove dangling coma here please.

housseindjirdeh replied:
I'm not too picky about dangling comments. On the contrary - I [prefer them](https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8).

I think it would be good to come to a consensus and enforce whichever way we want in the linter

alejandronanez replied:
Oh wow those are some good reasons to do dangling commas. I'm down with that :D

patw0929 replied:
I think maybe we can set up eslint & check lint rule before commit in next step to keep the coding style quality 😄

alejandronanez replied:
That's right, I was thinking about adding CircleCI or Travis in here, at least checking for eslint would be nice!

housseindjirdeh replied:
Opened an issue to discuss #66

alejandronanez commented on .eslintrc:
```diff
@@ -12,6 +12,10 @@
     ],
     "rules": {
-        // overrides
+        "no-console": 0
```
What about leaving `no-console` on and disable the line only when needed?

housseindjirdeh replied:
I'm assuming this allows `console.logs` in the code? Agreed, I would prefer to see it enforced actually

housseindjirdeh commented on root.store.js:
```diff
@@ -22,7 +31,20 @@ const getEnhancers = () => {
+export default function configureStore() {
+  const store = createStore(rootReducer, compose(getMiddleware(), ...getEnhancers()));
+
+  if (module.hot) {
```
Just curious, what exactly is this `if` condition for?

patw0929 replied:
I copied these redux devtools related code from my own project, but it seems unnecessary here, so I removed.

housseindjirdeh commented on root.store.js:
```diff
@@ -22,7 +31,20 @@ const getEnhancers = () => {
+    });
+  }
+
+  if (__DEV__) {
```
Do we need to save the `store` and `reducer` to the global object?

alejandronanez replied:
+1 on that question? What are you trying to do with this? I assume it is related with redux devtools but not 100% sure though.

patw0929 replied:
same reason, so I removed.

housseindjirdeh commented on .eslintrc:
```diff
@@ -12,6 +12,10 @@
+        "no-console": 0
+    },
+    "globals": {
+        "__DEV__": true,
```
👍

housseindjirdeh replied:
👍

alejandronanez commented:
I'm ok with what ever we choose for the search bar, the only restriction is that we should have only one implementation for iOS **and** Android, maintaining 2 components that do the same less maintainable and could ended up in a headache later down the road.

patw0929 commented:
Hey folks, thanks for reviewing & discussing this PR!
Sorry for the lately reply, I just woke up 😅
I had rebased master branch commits, fixed some conflicts, and fixed the code review advices above.

After thinking, about the iOS & Android SearchBar implementation, I agree with @alejandronanez, the maintaining cost might be a problem in the future.

alejandronanez commented on App.js:
```diff
@@ -26,6 +26,10 @@ import DeviceInfo from 'react-native-device-info';
 import md5 from 'md5';
 
+if (console) {
+  console.disableYellowBox = true; // eslint-disable-line no-console
```
👍

alejandronanez commented on README.md:
```diff
@@ -58,6 +58,7 @@ Feel free to send me feedback on [twitter](https://twitter.com/hdjirdeh) or [fil
 - `yarn run link` to link react-native dependencies
 - `yarn start` in one terminal tab to start the packager
 - `yarn run ios` in another tab to run the app in the iOS simulator
+- `yarn run android` in another tab to run the app in the Android device/emulator
```
❤️

alejandronanez commented on ios/GitPoint.xcodeproj/project.pbxproj:
```diff
@@ -36,6 +35,7 @@
 		2D02E4C91E0B4AEC006451C7 /* libReact.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 3DAD3EA31DF850E9000B6D8A /* libReact.a */; };
 		2DCD954D1E0B4F2C00145EB5 /* GitPointTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 00E356F21AD99517003FC87E /* GitPointTests.m */; };
 		3B4B2C15526143C288A991B2 /* libRNDeviceInfo.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 651F5DC3BF71489BB193A32B /* libRNDeviceInfo.a */; };
+		48F49F6A1F19028D0012FAD6 /* libRNSearchBar.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 48F49F671F19026D0012FAD6 /* libRNSearchBar.a */; };
```
I'm just curious about all changes in this file, are all this changes auto generated? if so, what triggered them?

patw0929 replied:
Because I added back the react-native-search-bar, which need added some files in Xcode.

housseindjirdeh replied:
Yep adding a native-bridged library will cause more than a few changes in native code base

alejandronanez replied:
Thanks for clarification =]

patw0929 commented:
Oops, stuck in some replacing problems:

1. When replacing with `react-native-elements`' SearchBar, I have no idea how to call `onCancel` event, and how to invoke the `unFocus` behavior.

2. If using `react-native-searchbar`, the SearchBar seems has absolute position, and behind the tabs.
![2017-07-15 2 44 45](https://user-images.githubusercontent.com/3055294/28237250-671e3800-696d-11e7-8698-f4822021a492.png)

alejandronanez commented:
> Okay in that case I'm okay with splitting the logic to two separate components for now - we can try consolidating them to a single component in a later ticket, don't want you to spend too much time trying to make it work with a single component that isn't rendering too well

Yeah agree, priority should be release android app ASAP. I created an issue for it https://github.com/gitpoint/git-point/issues/72 @housseindjirdeh I'm not sure where should it live in our project, can you please put it in there?

Also @patw0929 thanks for the awesome work and sorry for all these iterations!

housseindjirdeh commented:
@patw0929 really sorry if you had some back and forths with this - and I can't thank you enough for this PR. Just want to ask if you're okay with keeping the search bar components separate (I definitely am if the RNE solution/`react-native-searchbar` isn't working too well) and we can merge this beauty in :raised_hands:

patw0929 commented:
@housseindjirdeh Yes, I am ok with keeping search bar separated, and I think we can refine it in the next step. If everything works fine, please help to merge it 👍 Thanks!

housseindjirdeh commented:
Awesome @patw0929 :) 

Unfortunately some conflicts are present since we just merged #73, if you can please resolve them and I'll merge this baby in. (Don't hesitate to let me know if you would rather have me resolve them if you don't have the time :raised_hands: : )

alejandronanez commented:
Approving this since we already agreed on using 2 searchbars, keep console rules in eslint and better named variables. Just resolve merge conflicts @patw0929 and I think we should be good to merge.
